snippets/snippets.xml

```diff
@@ -0,0 +1,69 @@
+<variable name="FileName" expression="" defaultValue="" alwaysStopAt="true" />
+<variable name="MyComponent" expression="" defaultValue="" alwaysStopAt="true" />
+<context>
+    <option name="DART" value="true" />
+</context>
+</template>
+
+<template name="orCStful" value="import 'package:over_react/over_react.dart';&#10;&#10;// ignore: uri_has_not_been_generated&#10;part '$FileName$.over_react.g.dart';&#10;&#10;@Factory()&#10;// ignore: undefined_identifier&#10;UiFactory&lt;$MyComponent$Props&gt; Foo = $$$MyComponent$;&#10;&#10;@Props()&#10;class _$$$MyComponent$Props extends UiProps {}&#10;&#10;// AF-3369 This will be removed once the transition to Dart 2 is complete.&#10;// ignore: mixin_of_non_class, undefined_class&#10;class $MyComponent$Props extends _$$$MyComponent$Props with _$$$MyComponent$PropsAccessorsMixin {&#10;  // ignore: undefined_identifier, undefined_class, const_initialized_with_non_constant_value&#10;  static const PropsMeta meta = $$metaFor$MyComponent$Props;&#10;}&#10;&#10;@State()&#10;class _$$$MyComponent$State extends UiState {}&#10;&#10;// AF-3369 This will be removed once the transition to Dart 2 is complete.&#10;// ignore: mixin_of_non_class, undefined_class&#10;class $MyComponent$State extends _$$$MyComponent$State with _$$$MyComponent$StateAccessorsMixin {&#10;  // ignore: undefined_identifier, undefined_class, const_initialized_with_non_constant_value&#10;  static const StateMeta meta = $$metaFor$MyComponent$State;&#10;}&#10;&#10;@Component()&#10;class $MyComponent$Component extends UiStatefulComponent&lt;$MyComponent$Props, $MyComponent$State&gt; {&#10;// --------------------------------------------------------------------------&#10;// React Component Specifications and Lifecycle Methods&#10;// --------------------------------------------------------------------------&#10;&#10;  @override&#10;  Map getDefaultProps() =&gt; (newProps());&#10;&#10;  @override&#10;  Map getInitialState() =&gt; (newState());&#10;&#10;  @override&#10;  render() {}&#10;  &#10;// --------------------------------------------------------------------------&#10;// Private Utility Methods&#10;// --------------------------------------------------------------------------&#10;&#10;// --------------------------------------------------------------------------&#10;// Public Utility Methods&#10;// --------------------------------------------------------------------------&#10;&#10;// --------------------------------------------------------------------------&#10;// Public API Methods&#10;// --------------------------------------------------------------------------&#10;}" description="Creates a commented stateful over_react component" toReformat="false" toShortenFQNames="true">
```

Comment:
There's a `Foo` and `FooProps` in some of these templates that should use the template variable (`$MyComponent$`)